Packages/com.unity.multiplayer.samples.coop/Utilities/SceneManagement/SceneLoaderWrapper.cs

```diff
@@ -0,0 +1,146 @@
+        {
+            if (IsSpawned && IsNetworkSceneManagementEnabled && !NetworkManager.ShutdownInProgress)
+            {
+                if (NetworkManager.IsServer)
+                {
+                    // If is active server and NetworkManager uses scene management, load scene using NetworkManager's SceneManager
+                    NetworkManager.SceneManager.LoadScene(sceneName, loadSceneMode);
+                }
+            }
+            else
+            {
+                // If offline, load using SceneManager
+                var loadOperation = SceneManager.LoadSceneAsync(sceneName, loadSceneMode);
+                if (loadSceneMode == LoadSceneMode.Single)
+                {
+                    m_ClientLoadingScreen.StartLoadingScreen(sceneName, loadOperation);
+                }
+            }
+        }
+
+        void OnSceneLoaded(Scene scene, LoadSceneMode loadSceneMode)
+        {
+            if (!IsSpawned || NetworkManager.ShutdownInProgress)
+            {
+                m_ClientLoadingScreen.StopLoadingScreen();
+            }
+        }
+
+        void OnSceneEvent(SceneEvent sceneEvent)
+        {
+            switch (sceneEvent.SceneEventType)
+            {
+                case SceneEventType.Load: // Server told client to load a scene
+                    // Only executes on client
+                    if (NetworkManager.IsClient)
+                    {
+                        // Only start a new loading screen if scene loaded in Single mode, else simply update
+                        if (sceneEvent.LoadSceneMode == LoadSceneMode.Single)
+                        {
+                            m_ClientLoadingScreen.StartLoadingScreen(sceneEvent.SceneName, sceneEvent.AsyncOperation);
+                        }
+                        else
+                        {
+                            m_ClientLoadingScreen.UpdateLoadingScreen(sceneEvent.SceneName, sceneEvent.AsyncOperation);
+                        }
+                    }
+                    break;
+                case SceneEventType.LoadEventCompleted: // Server told client that all clients finished loading a scene
+                    // Only executes on client
+                    if (NetworkManager.IsClient)
+                    {
+                        m_ClientLoadingScreen.StopLoadingScreen();
+                    }
+                    break;
+                case SceneEventType.SynchronizeComplete: // Client told server that they finished synchronizing
+                    // Only executes on server
+                    if (NetworkManager.IsServer)
+                    {
+                        // Send client RPC to make sure the client stops the loading screen after the server handles what it needs to after the client finished synchronizing
+                        StopLoadingScreenClientRpc(new ClientRpcParams {Send = new ClientRpcSendParams {TargetClientIds = new[] {sceneEvent.ClientId}}});
```

Comment:
Why do we need to wait for the server to tell us to stop the loading screen? Do we have an event client side once we're done syncing? Why the back and forth?